recipes/dev/generate_v2.py

```diff
@@ -65,29 +66,37 @@ class InferenceRecipe:
 
     This *does not* currently support the following features:
         - torch.compile
-        - quantization through torchao
         - multi-GPU generation
         - batch generation
     """
 
     def __init__(self, cfg: DictConfig) -> None:
-        self._device = utils.get_device(device=cfg.device)
-        self._dtype = training.get_dtype(dtype=cfg.dtype, device=self._device)
-        self._logger = utils.get_logger(cfg.log_level)
+        self.device = utils.get_device(device=cfg.device)
+        self.dtype = training.get_dtype(dtype=cfg.dtype, device=self.device)
+        self.logger = utils.get_logger(cfg.log_level)
         training.set_seed(seed=cfg.seed)
 
     def setup(self, cfg: DictConfig) -> None:
         """Setup the model and transforms."""
-        # Load checkpointer and state_dict
+        # Load checkpointer
         _checkpointer = config.instantiate(cfg.checkpointer)
         _ckpt_dict = _checkpointer.load_checkpoint()
 
         # Instantiate model
-        with training.set_default_dtype(self._dtype), self._device:
+        with training.set_default_dtype(self.dtype), self.device:
             model = config.instantiate(cfg.model)
         model.load_state_dict(_ckpt_dict[training.MODEL_KEY])
+        self.logger.info(f"Model was initialized with precision {self.dtype}.")
+
+        # Quantize the model if specified
+        if cfg.get("quantization_method") is not None:
+            from torchao.quantization.quant_api import quantize_
+
+            quantization_method = config.instantiate(cfg.quantization_method)
+            compile_model(model)
```

Comment:
Compiling the model is necessary for quantization to be really worth it

Reply:
I'm curious whether compiling the model results in greater speedups than compiling the next-token-prediction fn like gptfast do

Reply:
we should compile after `quantize_` for speedup actually

Reply:
Interesting! I was following the pattern from AO's README where the model is compiled first:

```python
model = torchao.autoquant(torch.compile(model, mode='max-autotune'))
```

Why should the model be compiled after quantization?

Reply:
@jerryzh168 Anecdotally, I don't see much difference in tok/sec (after first token) between putting compile first or second. Can you share some more details about which one is correct?

Reply:
oh right now `quantize_` needs to compile after: https://github.com/pytorch/ao/blob/main/torchao/quantization/README.md#full-affine-quantization-flow-example

but `autoquant` will do compile first before calling `autoquant`

Reply:
mode='max-autotune' will take a long time to compile. Is it worth it? We dont do it for training.

Its interesting that in AO's read it says to put compile first. Do we also do it for QLoRA?

Reply:
> I don't see much difference in tok/sec (after first token) between putting compile first or second.

I haven't tried calling `quantize_` after compile actually, maybe it would have the same effect as well, need to confirm

Reply:
> Its interesting that in AO's read it says to put compile first. Do we also do it for QLoRA?

This is only for autoquant actually, we also haven't tested QLoRA, I think @andrewor14 is taking a look now

If compile time is a concern, we are also thinking of just do autoquant before hand and save the model, but I'm still testing that path as well